### ISSUE TITLE
Hyperparameters 📚 

### ISSUE THREAD

**Issue by ivankunyankin** [ivankunyankin/quartznet-asr#2]
A short description of hyperparameters from config.yml

### Training parameters
1. ```max_length``` - maximum file length in seconds. Longer files are omitted. Used to avoid OOM (out-of-memory) error. Also reduces the amount of data samples used for training and validation.
2. ```weight_decay``` - regularization parameter. Can be increased when using constant learning rate
3. ```use_onecyclelr``` - use PyTorch's [implementation](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.OneCycleLR.html#torch.optim.lr_scheduler.OneCycleLR) of cyclic learning rate policy
4. ```spec_params``` - a set of parameters for generating mel spectrograms. Described [here](https://librosa.org/doc/main/generated/librosa.feature.melspectrogram.html). When changing these one should consider changing parameters of masking. 
5. ```normalize``` - apply normalization. By default applies normalization from stats saved in ```assests/stats.npy``` to each mel channel separately. Saved stats were calculated from LibriTTS. If training on different data, these should be recalculated. More details [here](https://github.com/ivankunyankin/quartznet-asr/issues/1).

### Augmentation parameters
1. ```speed_perturbation``` - randomly chooses the rate of perturbation between 1 +- specified value. 
2. ```chunk_size``` -  size of each spectrogram segment (time axis) to apply augmentation to. Set equal to -1 to apply original pytorch's masking. ```Examples shown below```
3. ```freq_masking, time_masking``` - the maximum width of masking band. More details [here](https://pytorch.org/audio/stable/transforms.html#frequencymasking)

### Augmentation examples
<div align="center"><i><small>Original pytorch functions applied. Less aggresive augmentation</small></i></div>
<img width="800" alt="Screenshot 2021-07-01 at 11 37 33" src="https://user-images.githubusercontent.com/47857592/124658791-dee19900-deac-11eb-9a9c-92a30c107406.png">

<div align="center"><i><small>Custom function applied. More aggresive augmentation</small></i></div>
<img width="800" alt="Screenshot 2021-07-01 at 11 38 00" src="https://user-images.githubusercontent.com/47857592/124658817-e9039780-deac-11eb-92e4-d5f5053b9c64.png">
